wicker/core/config.py

```diff
@@ -29,22 +42,31 @@ def from_json(cls, data: Dict[str, Any]) -> WickerAwsS3Config:
 class WickerConfig:
     raw: Dict[str, Any]
     aws_s3_config: WickerAwsS3Config
+    wandb_config: WickerWandBConfig
 
     @classmethod
     def from_json(cls, data: Dict[str, Any]) -> WickerConfig:
         return cls(
             raw=data,
             aws_s3_config=WickerAwsS3Config.from_json(data["aws_s3_config"]),
+            wandb_config=WickerWandBConfig.from_json(data.get("wandb_config", {})),
         )
 
 
 _CONFIG: Optional[WickerConfig] = None
+_PREV_WICKER_CONFIG_PATH: Optional[str] = None
 
 
 def get_config() -> WickerConfig:
     """Retrieves the Wicker config for the current process"""
     global _CONFIG
-    if _CONFIG is None:
-        with open(WICKER_CONFIG_PATH, "r") as f:
+    global _PREV_WICKER_CONFIG_PATH
+
+    loc_wicker_config_path = os.getenv("WICKER_CONFIG_PATH", os.path.expanduser("~/wickerconfig.json"))
+    # Hack to get around having tests need to specify config hermetically
+    # ToDo: Abstract the config additionally to clean this up
+    if _CONFIG is None or loc_wicker_config_path != _PREV_WICKER_CONFIG_PATH:
+        _PREV_WICKER_CONFIG_PATH = loc_wicker_config_path
+        with open(loc_wicker_config_path, "r") as f:
```

Comment:
This supports both one time only loading and changing configs for tests. The way I specified the wandb tests required changing configs. It also supports having better more hermetic unit tests down the road if we abstract the strict static config away from the tests.

Reply:
Yeah that was my thought too, where I was mocking it out in this block, https://github.com/woven-planet/wicker/blob/feature/isaak_willett/wandb_dataset_versioning/tests/test_wandb.py#L14-L31. The problem was that the global on L62 prevented us from changing the config on loading. So it only got set once when the raw code was parsed the first time and set to a global. I'm assuming the global was there to not have the I/O lag from loading the config multiple times but it allowed us to only ever load one config even for testing.
The idea here was if the path changed we would reload but it could have ramifications down the road with unexpected behavior. I wasn't sure a good solution to this since I didn't like having all the tests be tied together and only allow one test per function that was config dependent.

Reply:
That was a word salad, basically it was to get around the hard requirement before of only having one config without breaking the only loading the config once. It does mock out the config it just can't be loaded again without this if we keep the global.